TidepoolKit/Internal/Responses/DataResponse.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  DataResponse.swift
+//  TidepoolKit
+//
+//  Created by Darin Krauss on 2/28/20.
+//  Copyright © 2020 Tidepool Project. All rights reserved.
+//
+
+struct DataResponse: Codable {
+    var data: [TDatum]
```

Comment:
Good question. 

One thing I struggled with was dealing with all (17) of the properties common to all Tidepool data types (as implemented in the TDatum base class). If I implemented properties as `let` then every single property would have to be specified in every initializer for all Tidepool data types. This got very unreadable and annoying very fast. 

I decided the ability to create the object using a straightforward, type-specific constructor (with only the few type-specific properties) and then allowing modification of the objects afterwards (from `var`) to set any of the infrequently-used common properties made the most sense.

In addition, while writing code to use these classes, I realized that there is the added benefit of being able to set these common properties en-mass after creating a number of data.

While it is not ideal I think this is a reasonable compromise.

Reply:
I admit that adding all those initializers is a drag.  In the past, I've done exactly what you've done: started with `var`s because it was such a pain.  In the end, I eventually came around and realized that immutability is actually more beneficial to the users, and it was just boilerplate code anyway.  That said, not a blocker for this review, but something to chew on...